### PR TITLE
Adds reduce, maybe and toArray to Maybe

### PR DESCRIPTION
**Pull request by scott-christopher**
Example:
```js
// Maybe.maybe :: b -> (a -> b) -> Maybe a -> b
Maybe.maybe(42, x => 2 * x, Just(10));  // 20
Maybe.maybe(42, x => 2 * x, Nothing()); // 42

// Maybe.toArray :: Maybe a -> [a]
Maybe.toArray(Just(42));  // [42]
Maybe.toArray(Nothing()); // []

// Maybe m => m.reduce :: (z -> a -> z) -> z -> z
Just(32).reduce(R.add, 10);  // 42
Nothing().reduce(R.add, 10); // 10
```

### CODE REVIEW

**Comment by buzzdecafe** [comment]
:cow2: good stuff

**Comment by TheLudd** [comment]
LGTM as long as the `Nothing` cases are correctly implemented. I find this hard to judge sometimes. 

But looks as if we need to add `Foldable` to the table in the readme. Could you do that as well @scott-christopher?

**Comment by davidchambers** [comment]
:thumbsup:

File: src/Maybe.js [hunk 1]
```diff
@@ -36,6 +36,18 @@ Maybe.isNothing = function(x) {
   return x === _nothing;
 };
 
+Maybe.toArray = function(m) {
```
Comment (joneshf) [inline src/Maybe.js]: You should be able to derive this once and for all for any `Foldable` as:

```js
function toArray(x) {
  return x.reduce(R.flip(R.append), []);
}
```

Comment (scott-christopher) [inline src/Maybe.js]: I thought about that, but wasn't sure where it should go.

Is it worth introducing a separate file that can hold functions that apply to `Foldable` instances? It seems a bit too fantasyland focused to end up in Ramda proper.

Comment (buzzdecafe) [inline src/Maybe.js]: a separate file would be fine, or we have a grab-bag of utility functions here: https://github.com/ramda/ramda-fantasy/blob/master/src/internal/util.js

Comment (scott-christopher) [inline src/Maybe.js]: I've removed `toArray` from this PR and I'll raise a new PR in a few days time to include a separate `Foldable.js` with a handful of useful functions that operate of foldable types.

**Comment by scott-christopher** [comment]
@TheLudd, I've updated the PR to include a Foldable column to the README.

**Pull request merged**